### PR TITLE
fix: fixes the memory leak for volumes

### PR DESCRIPTION
**Pull request by sedghi**
This fixes some of the memory issues in which vtk's render window interactor was not unsubscribing from one of its events. 
Right now memory profiling in Chrome does not show any noticeable objects left for decaching but still the demos keep some GPU process. 

We might want to investigate more on this

### CODE REVIEW

**Review by wayfarer3130 — APPROVED** [review]
Looks good to me - only see if you want to bind the one item.  I don't htink you need to, but I wanted you to check.

File: packages/core/src/RenderingEngine/vtkClasses/vtkSharedVolumeMapper.js [hunk 1]
```diff
@@ -14,6 +14,12 @@ import vtkVolumeMapper from '@kitware/vtk.js/Rendering/Core/VolumeMapper';
  */
 function vtkSharedVolumeMapper(publicAPI, model) {
   model.classHierarchy.push('vtkSharedVolumeMapper');
+
+  const superDelete = publicAPI.delete;
+  publicAPI.delete = () => {
+    model.scalarTexture = null;
+    superDelete();
```
[the remark this answers is not in the document]
Comment (sedghi) [reply packages/core/src/RenderingEngine/vtkClasses/vtkSharedVolumeMapper.js:21]: there is no this in vtk.js, it is all functional, so it figures it out

**Comment by Ouwen** [comment]
@sedghi I was wondering if there was any resolution to the GPU memory process lingering. When I'm in OHIF and switch modes the GPU process memory footprint creeps up even with the destroy and forced loss of webGL context.

I'm wondering if there is something else which is lingering.

**Comment by sedghi** [comment]
@Ouwen can you create an issue and we discuss there?

**Comment by Ouwen** [comment]
Sure thing

**Pull request merged**